gufe/protocols/protocoldag.py

```diff
@@ -385,52 +450,92 @@ def execute_DAG(protocoldag: ProtocolDAG, *,
         The result of executing the `ProtocolDAG`.
 
     """
+    # the directory given as shared_root is actually the directory for this
+    # DAG; the "shared_root" for the storage manager is the parent. We'll
+    # force permanent to be the same.
+    storage_manager = ReproduceOldBehaviorStorageManager.from_old_args(
+        shared_basedir=shared_basedir,
+        scratch_basedir=scratch_basedir,
+        keep_shared=keep_shared,
+        keep_scratch=keep_scratch
+    )
+    dag_label = shared_basedir.name
+    return new_execute_DAG(protocoldag, dag_label, storage_manager,
+                           raise_error, n_retries)
+
+
+def new_execute_DAG(  # TODO: this is a terrible name
+    protocoldag: ProtocolDAG,
+    dag_label: str,
```

Comment:
why is a DAG label now required?

and why can't the label just be taken off the input DAG?

Reply:
Breaks current behavior, where the equivalent of the DAG label is given by the user as the `-o` from quickrun.